Autopilot/PathManager/Src/landingTakeoffManager.cpp

```diff
@@ -0,0 +1,204 @@
+    Vector3D sPoint(stoppingPoint.longitude, stoppingPoint.latitude, stoppingPoint.altitude); //stopping point
+    Vector3D cPoint(input.longitude, input.latitude, input.altitude); //current point
+    Vector3D projectedPoint;
+
+    //calculating the vectors
+    Vector3D vectorAI(iPoint.x - aPoint.x, iPoint.y - aPoint.y, iPoint.z - aPoint.z);
+    Vector3D vectorAS(sPoint.x - aPoint.x, sPoint.y - aPoint.y, sPoint.z - aPoint.z);
+    Vector3D vectorAC(cPoint.x - aPoint.x, cPoint.y - aPoint.y, cPoint.z - aPoint.z);
+    Vector3D normalVector;
+    
+    //normal of the plane created by aiming, stopping, and intersection point
+    normalVector = vectorAI.crossProduct(vectorAS);
+
+    //projecting vectorAC to the normal
+    //equation is the dot product of normal and vectorAC, divided by the norm of the normal squared, multiplied by the normal vector
+    Vector3D projectionACToNormal;
+
+    projectionACToNormal = normalVector * (normalVector.dotProduct(vectorAC) / pow(normalVector.norm(),2));
+
+    //project point onto plane (cpoint minus the projection)
+
+    projectedPoint = cPoint - projectionACToNormal;
+
+    //determine the equation of the line of the slope and sub in projected values (symmetric form)
+    double altitude = (((projectedPoint.x - aPoint.x) / vectorAI.x) + aPoint.z) * vectorAI.z;
+
+    return altitude;
+}
+
+double LandingManager::approachSpeed(bool ifPackage)
+{
+    //approach speed calculation for both package and no package scenarios
+    if(ifPackage)
+    {
+        return STALL_SPEED_WITH_PACKAGE * 1.3;
+    }
+    else 
+    {
+        return STALL_SPEED_NO_PACKAGE * 1.3;
+    }
+}
+
+double LandingManager::slowFlightSpeed(bool ifPackage)
+{
+    //slow flight speed calculation for both package and no package scenarios
+    if(ifPackage)
+    {
+        return STALL_SPEED_WITH_PACKAGE + 2;
+    }
+    else 
+    {
+        return STALL_SPEED_NO_PACKAGE + 2;
+    }
+}
+
+_LandingPath LandingManager::createSlopeWaypoints(Telemetry_PIGO_t input, double currentAltitude)
+{
+    //created comments for testing
+    _LandingPath path;
+    path.stoppingPoint.latitude = input.stoppingLatitude; //0
+    path.stoppingPoint.longitude = input.stoppingLongitude; //0
+    path.stoppingPoint.altitude = input.stoppingAltitude; //0
+
+    //creating points from stopping point to aiming point 
+    
+    //setting Z of aiming point
+    path.aimingPoint.altitude = input.stoppingAltitude; //0
+    
+    //determining x and y of aiming point
+    double radianDirection = input.stoppingDirectionHeading * PI / 180.0; //PI/4
+
+    //finding the x and y components of the rolling distance vector
+    double stoppingDistX = sin(radianDirection) * DISTANCE_OF_LANDING; //5root2
+    double stoppingDistY = cos(radianDirection) * DISTANCE_OF_LANDING; //5root2
+
+    //converting into x and y components in lat and lon
+    double metersPerDegLon = 40075000.0 * cos(input.stoppingLatitude*PI/180)/360.0; //111251.6316
```

Comment:
would it make sense to make 40075000.0 a constexpr?

Reply:
sure I can do that!